Wikipedia/Code/WMFArticleFetcher.m

```diff
@@ -99,13 +99,25 @@ - (nullable NSURLSessionTask *)fetchArticleForURL:(NSURL *)articleURL
         failure([NSError wmf_errorWithType:WMFErrorTypeStringMissingParameter userInfo:nil]);
         return nil;
     }
-
+    
     // Force desktop domain if not Zero rated.
     if (![SessionSingleton sharedInstance].zeroConfigurationManager.isZeroRated) {
         useDeskTopURL = YES;
     }
 
     NSURL *url = useDeskTopURL ? [NSURL wmf_desktopAPIURLForURL:articleURL] : [NSURL wmf_mobileAPIURLForURL:articleURL];
+    
+    __block WMFErrorHandler saveToDiskFailure = nil;
+    if (saveToDisk) {
+        saveToDiskFailure = ^(NSError *error) {
+            if (error.domain == NSCocoaErrorDomain && error.code == NSFileWriteOutOfSpaceError) {
```

Comment:
Does this error stop other disk save operations from being attempted? I'm wondering if the user will just see like 100 of these alerts in a row.

Reply:
Also, does the app keep trying to download articles/images even though saves are failing? IOW will the app keep trying to re-download the same articles over an over again only to have saves fail?

We are getting into edge cases here, but something to think about.

Reply:
Yup, the way it's implemented now, both are true. The error should be displayed only when user attempts to save via the UI so I should modify this PR and pass that failure block only when that happens (right now, I'm passing it in the wrong place and I end up with an error anytime save happens, regardless of whether it was initiated by user action). I should check with Carolyn how we want to handle the 'your account has more articles than your device can handle' case, maybe that should add another label on the article cell rather than an alert